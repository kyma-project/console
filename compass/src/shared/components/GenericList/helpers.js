export const filterEntries = (entries, query) => {
  // TEMP
  if (
    entries[0] &&
    entries[0].hasOwnProperty('name') &&
    entries[0].hasOwnProperty('description')
  ) {
    return entries.filter(entry => {
      return (
        entry.name.indexOf(query) > -1 || entry.description.indexOf(query) > -1
      );
    });
  } else {
    return entries;
  }
};
