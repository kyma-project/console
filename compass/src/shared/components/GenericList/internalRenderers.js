import React from 'react';
import { Popover } from 'fundamental-react/lib/Popover';
import { Menu } from 'fundamental-react/lib/Menu';
import { Button } from '@kyma-project/react-components';

export const renderActionElement = (actions, entry) =>
  Array.isArray(actions) &&
  ((
    <Popover
      body={
        <Menu>
          <Menu.List>
            {actions.map((
              action,
              id, // no unique key error appears here. 'key' is not passed further by Fd-react
            ) => (
              <Menu.Item
                onClick={() => action.handler(entry)}
                key={(entry.name || entry.id) + id}
              >
                {action.name}
              </Menu.Item>
            ))}
          </Menu.List>
        </Menu>
      }
      control={<Button glyph="vertical-grip" option="light" />}
      placement="bottom-end"
    />
  ) ||
    null);
