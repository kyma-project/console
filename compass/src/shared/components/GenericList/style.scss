.generic-list-search {
    .fd-menu__list {
        max-height: 50vh;
        overflow-y: scroll;

        .fd-menu__item {
            overflow: hidden;
            text-overflow: ellipsis;
        }

        .fd-menu__item--no-entries {
            font-style: italic;
        }
    }
}
