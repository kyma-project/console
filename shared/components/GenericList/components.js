import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { Button } from 'fundamental-react';

import ListActions from '../ListActions/ListActions';

export const BodyFallback = ({ children }) => (
  <tr>
    <td colSpan="100%">
      <div className="body-fallback">{children}</div>
    </td>
  </tr>
);

export const HeaderRenderer = ({ actions, headerRenderer }) => {
  let emptyColumn = [];
  if (actions.length) {
    emptyColumn = [<th key="actions-column" aria-label="actions-column"></th>];
  }
  return [
    headerRenderer().map((h, index) => <th key={h || index}>{h}</th>),
    ...emptyColumn,
  ];
};

export const RowRenderer = ({ entry, actions, rowRenderer, ...others }) => {
  const filteredActions = actions.filter(a =>
    a.skipAction ? !a.skipAction(entry) : true,
  );
  const resolvedRowRenderer = rowRenderer(entry);

  if (Array.isArray(resolvedRowRenderer)) {
    return (
      <DefaultRowRenderer
        {...others}
        entry={entry}
        actions={filteredActions}
        rowRenderer={resolvedRowRenderer}
      />
    );
  }
  return (
    <CollapsedRowRenderer
      {...others}
      entry={entry}
      actions={filteredActions}
      rowRenderer={resolvedRowRenderer}
    />
  );
};

const DefaultRowRenderer = ({
  key,
  entry,
  actions,
  rowRenderer,
  actionsStandaloneItems,
}) => {
  const cells = rowRenderer.map((cell, id) => <td key={id}>{cell}</td>);
  const actionsCell = (
    <td>
      <ListActions
        actions={actions}
        entry={entry}
        standaloneItems={actionsStandaloneItems}
      />
    </td>
  );

  return (
    <tr key={key}>
      {cells}
      {!!actions.length && actionsCell}
    </tr>
  );
};

const CollapsedRowRenderer = ({
  key,
  entry,
  actions,
  rowRenderer: { cells, collapseContent, showCollapseControl = true },
  actionsStandaloneItems,
}) => {
  const [isOpen, setOpen] = useState(false);

  const rowRenderer = [
    showCollapseControl ? (
      <Button
        data-testid={isOpen ? 'collapse-button-open' : 'collapse-button-close'}
        glyph={isOpen ? 'navigation-up-arrow' : 'navigation-down-arrow'}
        option="light"
        onClick={() => setOpen(!isOpen)}
        compact={true}
        typeAttr="button"
      />
    ) : (
      <></>
    ),
    ...cells,
  ];

  const defaultRow = (
    <DefaultRowRenderer
      key={key}
      entry={entry}
      actions={actions}
      rowRenderer={rowRenderer}
      actionsStandaloneItems={actionsStandaloneItems}
    />
  );

  return (
    <>
      {defaultRow}
      {isOpen ? (
        <tr role="row" data-testid="collapse-content">
          {collapseContent}
        </tr>
      ) : null}
    </>
  );
};
