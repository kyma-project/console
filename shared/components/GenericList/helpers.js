const filterEntry = (entry, query, searchProperties) => {
  if (!query) {
    return true;
  }

  if (!Object.keys(searchProperties).length) {
    return false;
  }
  for (const property of searchProperties) {
    if (entry.hasOwnProperty(property)) {
      const value = entry[property];
      // apply to string to include numbers
      if (
        value &&
        value
          .toString()
          .toLowerCase()
          .indexOf(query.toLowerCase()) !== -1
      ) {
        return true;
      }
    }
  }
  return false;
};

export const filterEntries = (entries, query, searchProperties) => {
  return entries.filter(entry => filterEntry(entry, query, searchProperties));
};
