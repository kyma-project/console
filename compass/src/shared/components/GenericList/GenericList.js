import React from 'react';
import PropTypes from 'prop-types';

import { Panel } from 'fundamental-react/lib/Panel';

import {
  Search,
  TableWithActionsToolbar,
  TableWithActionsList,
} from '@kyma-project/react-components';

import { filterEntries } from './helpers';
import { renderActionElement } from './internalRenderers';

class GenericList extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      entries: props.entries,
      filteredEntries: props.entries,
    };
  }

  headerRenderer = entries => {
    if (this.props.actions) {
      return [...this.props.headerRenderer(entries), ''];
    } else {
      return this.props.headerRenderer(entries);
    }
  };

  rowRenderer = entry => {
    if (this.props.actions) {
      return [
        ...this.props.rowRenderer(entry),
        renderActionElement(this.props.actions, entry),
      ];
    } else {
      return this.props.rowRenderer(entry);
    }
  };

  renderSearchList = entries => {
    const filteredEntries =
      entries && entries.length
        ? entries.map(entry => {
            return entry.name !== undefined && entry.description !== undefined
              ? [
                  {
                    text: entry.name.substring(0, 18),
                    callback: () => this.handleQueryChange(entry.name),
                  },
                  {
                    text: entry.description.substring(0, 18),
                    callback: () => this.handleQueryChange(entry.description),
                  },
                ]
              : entries;
          })
        : null;
    return Array.isArray(filteredEntries) ? filteredEntries.flat() : entries;
  };

  handleQueryChange = event => {
    const searchTerm = event.target.value;
    this.setState(prevState => ({
      filteredEntries: filterEntries(prevState.entries, searchTerm),
    }));
  };

  render() {
    const { filteredEntries } = this.state;
    const { extraHeaderContent } = this.props;

    const headerActions = filteredEntries => (
      <>
        {extraHeaderContent}
        {/* {this.processFilterElement(allFilters)} */}
        <Search
          placeholder="Search..."
          onChange={this.handleQueryChange}
          searchList={this.renderSearchList(filteredEntries)}
        />
      </>
    );

    return (
      <Panel className="fd-panel--no-background">
        <TableWithActionsToolbar
          title={this.props.title}
          description={this.props.description}
          children={headerActions(filteredEntries)}
        />

        <Panel.Body>
          <TableWithActionsList
            entries={filteredEntries}
            headerRenderer={this.headerRenderer}
            rowRenderer={this.rowRenderer}
          />
        </Panel.Body>
      </Panel>
    );
  }
}
export default GenericList;

GenericList.propTypes = {
  title: PropTypes.string.isRequired,
  description: PropTypes.string,
  entries: PropTypes.arrayOf(PropTypes.object),
  headerRenderer: PropTypes.func.isRequired,
  rowRenderer: PropTypes.func.isRequired,
  actions: PropTypes.arrayOf(
    PropTypes.shape({ name: PropTypes.string, handler: PropTypes.func }),
  ),
  extraHeaderContent: PropTypes.node,
};
