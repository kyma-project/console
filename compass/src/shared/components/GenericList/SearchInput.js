import React from 'react';
import PropTypes from 'prop-types';
import './style.scss';
import 'core-js/es/array/flat-map';

SearchInput.propTypes = {
  searchQuery: PropTypes.string,
  entries: PropTypes.arrayOf(PropTypes.object.isRequired),
  handleQueryChange: PropTypes.func.isRequired,
};

export default function SearchInput({
  searchQuery,
  filteredEntries,
  handleQueryChange,
}) {
  const [isSearchHidden, setSearchHidden] = React.useState(true);
  const searchInputRef = React.useRef();

  const renderSearchList = entries => {
    const suggestions = getSearchSuggestions(entries);

    if (!suggestions.length) {
      return (
        <li
          key="no-entries"
          className="fd-menu__item fd-menu__item--no-entries"
        >
          No entries found
        </li>
      );
    }

    return suggestions.map(suggestion => (
      <li
        onClick={() => handleQueryChange(suggestion)}
        key={suggestion}
        className="fd-menu__item"
      >
        {suggestion}
      </li>
    ));
  };

  const getSearchSuggestions = entries => {
    const suggestions = entries
      .flatMap(entry => {
        const headers = ['name', 'description'];
        return headers.map(header => {
          const entryValue = entry[header];
          if (entryValue && entryValue.includes(searchQuery)) {
            return entryValue;
          }
          return null;
        });
      })
      .filter(suggestion => suggestion);
    return Array.from(new Set(suggestions));
  };

  const openSearchList = () => {
    const inputField = searchInputRef.current;
    if (inputField !== document.activeElement) {
      inputField.focus();
    }
  };

  return (
    <div className="fd-search-input generic-list-search">
      <div className="fd-popover">
        <div className="fd-popover__control">
          <div className="fd-combobox-control">
            <div className="fd-input-group fd-input-group--after">
              <input
                ref={searchInputRef}
                type="text"
                className="fd-input"
                placeholder="Search..."
                value={searchQuery}
                onChange={e => handleQueryChange(e.target.value)}
                onFocus={() => setSearchHidden(false)}
                onBlur={() => setSearchHidden(true)}
              />
              <span className="fd-input-group__addon fd-input-group__addon--after fd-input-group__addon--button">
                <button
                  className="fd-button--light sap-icon--search"
                  onClick={openSearchList}
                ></button>
              </span>
            </div>
          </div>
        </div>
        <div
          className="fd-popover__body fd-popover__body--no-arrow"
          aria-hidden={isSearchHidden}
        >
          <nav className="fd-menu">
            <ul className="fd-menu__list">
              {renderSearchList(filteredEntries)}
            </ul>
          </nav>
        </div>
      </div>
    </div>
  );
}
