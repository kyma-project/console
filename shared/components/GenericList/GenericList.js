import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import SearchInput from './SearchInput';
import { Panel } from 'fundamental-react/Panel';
import { filterEntries } from './helpers';
import ListActions from '../ListActions/ListActions';
import './GenericList.scss';

const NotFoundMessage = ({ children }) => (
  <td colSpan="100%">
    <p className="not-found-message">{children}</p>
  </td>
);

const HeaderRenderer = ({ entries, actions, headerRenderer }) => {
  let emptyColumn = [];
  if (actions.length) {
    emptyColumn = [<th key="actions-column" aria-label="actions-column"></th>];
  }
  return [headerRenderer().map(h => <th key={h}>{h}</th>), ...emptyColumn];
};

const RowRenderer = ({
  entry,
  actions,
  rowRenderer,
  actionsStandaloneItems,
}) => {
  const filteredActions = actions.filter(a =>
    a.skipAction ? !a.skipAction(entry) : true,
  );

  const cells = rowRenderer(entry).map((cell, id) => <td key={id}>{cell}</td>);
  const actionsCell = (
    <td>
      <ListActions
        actions={filteredActions}
        entry={entry}
        standaloneItems={actionsStandaloneItems}
      />
    </td>
  );

  return (
    <>
      {cells}
      {!!filteredActions.length && actionsCell}
    </>
  );
};

export const GenericList = ({
  entries,
  actions,
  title,
  headerRenderer,
  rowRenderer,
  notFoundMessage,
  extraHeaderContent,
  showSearchField,
  textSearchProperties,
  actionsStandaloneItems,
}) => {
  const [filteredEntries, setFilteredEntries] = useState(entries);
  const [searchQuery, setSearchQuery] = useState('');

  useEffect(() => {
    setFilteredEntries(
      filterEntries([...entries], searchQuery, textSearchProperties),
    );
  }, [searchQuery, setFilteredEntries, entries]);

  const headerActions = (
    <section className="generic-list__search">
      {showSearchField && (
        <SearchInput
          searchQuery={searchQuery}
          filteredEntries={filteredEntries}
          handleQueryChange={setSearchQuery}
          suggestionProperties={textSearchProperties}
        />
      )}
      {extraHeaderContent}
    </section>
  );

  return (
    <Panel className="fd-has-margin-m generic-list">
      <Panel.Header className="fd-has-padding-xs">
        <Panel.Head title={title} />
        <Panel.Actions>{headerActions}</Panel.Actions>
      </Panel.Header>

      <Panel.Body>
        <table className="fd-table">
          <thead>
            <tr>
              <HeaderRenderer
                entries={entries}
                actions={actions}
                headerRenderer={headerRenderer}
              />
            </tr>
          </thead>
          <tbody>
            {filteredEntries.length ? (
              filteredEntries.map((e, index) => (
                <tr role="row" key={e.id || e.name || index}>
                  <RowRenderer
                    entry={e}
                    actions={actions}
                    actionsStandaloneItems={actionsStandaloneItems}
                    rowRenderer={rowRenderer}
                  />
                </tr>
              ))
            ) : (
              <tr>
                <NotFoundMessage>{notFoundMessage}</NotFoundMessage>
              </tr>
            )}
          </tbody>
        </table>
      </Panel.Body>
    </Panel>
  );
};

GenericList.Actions = ListActions;

GenericList.propTypes = {
  title: PropTypes.string,
  entries: PropTypes.arrayOf(PropTypes.object).isRequired,
  headerRenderer: PropTypes.func.isRequired,
  rowRenderer: PropTypes.func.isRequired,
  actions: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string,
      handler: PropTypes.func.isRequired,
      skipAction: PropTypes.func,
    }),
  ).isRequired,
  extraHeaderContent: PropTypes.node,
  showSearchField: PropTypes.bool,
  notFoundMessage: PropTypes.string,
  textSearchProperties: PropTypes.arrayOf(PropTypes.string.isRequired),
  actionsStandaloneItems: PropTypes.number,
};

GenericList.defaultProps = {
  notFoundMessage: 'No entries found',
  actions: [],
  showSearchField: true,
  textSearchProperties: ['name', 'description'],
};
