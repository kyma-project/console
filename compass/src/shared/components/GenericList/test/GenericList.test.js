import React from 'react';
import GenericList from '../GenericList';
import renderer from 'react-test-renderer';
import 'core-js/es/array/flat-map';

describe('GenericList', () => {
  const mockHeaderRenderer = entries => ['Name'];
  const mockEntryRenderer = entry => [entry.name];
  const mockEntries = [
    { name: 'first_entry', description: 'testdescription1' },
    { name: 'second_entry', description: 'testdescription2' },
  ];

  it('Renders with minimal props', () => {
    const component = renderer.create(
      <GenericList
        title=""
        entries={[]}
        headerRenderer={() => []}
        rowRenderer={() => []}
      />,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Renders title and description', () => {
    const [title, description] = ['title', 'description'];
    const component = renderer.create(
      <GenericList
        title={title}
        description={description}
        entries={[]}
        headerRenderer={() => []}
        rowRenderer={() => []}
      />,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Renders actions button when 'actions' prop is provided", () => {
    const actions = [{ name: 'testaction', handler: () => {} }];

    const component = renderer.create(
      <GenericList
        title=""
        actions={actions}
        entries={mockEntries}
        headerRenderer={mockHeaderRenderer}
        rowRenderer={mockEntryRenderer}
      />,
    );
    let tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("Skips rendering actions when 'actions' prop passes skipAction() call", () => {
    let actions = [
      { name: 'testaction', handler: () => {}, skipAction: () => true },
    ];

    let component = renderer.create(
      <GenericList
        title=""
        actions={actions}
        entries={mockEntries}
        headerRenderer={mockHeaderRenderer}
        rowRenderer={mockEntryRenderer}
      />,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
