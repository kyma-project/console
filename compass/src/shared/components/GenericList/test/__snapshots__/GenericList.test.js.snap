// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenericList Renders actions button when 'actions' prop is provided 1`] = `
<div
  className="fd-panel fd-panel--no-background"
>
  <div
    className="fd-action-bar sc-kafWEX kXuvQ"
    data-e2e-id="toolbar"
  >
    <div
      className="fd-action-bar__header sc-iELTvK cWSWPl"
      data-e2e-id="toolbar-header"
    >
      <h1
        className="fd-action-bar__title"
      >
        
      </h1>
    </div>
    <div
      className="fd-action-bar__actions sc-cmTdod fVngQh"
      data-e2e-id="toolbar-actions"
    >
      <div
        className="fd-search-input"
      >
        <div
          className="fd-popover"
        >
          <div
            className="fd-popover__control"
          >
            <div
              aria-expanded={false}
              aria-haspopup="true"
              className="fd-combobox-control"
            >
              <div
                className="fd-input-group fd-input-group--after"
              >
                <input
                  className="fd-input"
                  onChange={[Function]}
                  onClick={[Function]}
                  onKeyPress={[Function]}
                  placeholder="Type your query"
                  style={Object {}}
                  type="text"
                  value=""
                />
                <span
                  className="fd-input-group__addon fd-input-group__addon--after fd-input-group__addon--button"
                >
                  <button
                    className="fd-button--light sap-icon--search"
                    onClick={[Function]}
                  />
                </span>
              </div>
            </div>
          </div>
          <div
            aria-hidden={true}
            className="fd-popover__body fd-popover__body--no-arrow"
          >
            <div
              className=""
            >
              <nav
                className="fd-menu"
              >
                <ul
                  className="fd-menu__list"
                >
                  <li
                    className="fd-menu__item"
                    onClick={[Function]}
                  >
                    <strong>
                      
                    </strong>
                  </li>
                  <li
                    className="fd-menu__item"
                    onClick={[Function]}
                  >
                    <strong>
                      
                    </strong>
                  </li>
                </ul>
              </nav>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="fd-panel__body abc"
  >
    <div
      className="fd-panel sc-gGBfsJ eaQlFN"
    >
      <div
        className="fd-panel__body sc-hEsumM gUkXwJ"
      >
        <table
          className="fd-table sc-ktHwxA kmRrJU"
        >
          <thead>
            <tr>
              <th>
                Name
              </th>
              <th>
                
              </th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>
                first_entry
              </td>
              <td>
                <div
                  className="fd-popover"
                >
                  <div
                    onFocus={[Function]}
                    onMouseDown={[Function]}
                    onTouchStart={[Function]}
                  >
                    <div
                      className="fd-popover__control"
                    >
                      <button
                        className="fd-button--light sap-icon--vertical-grip"
                        onClick={[Function]}
                      />
                    </div>
                  </div>
                </div>
              </td>
            </tr>
            <tr>
              <td>
                second_entry
              </td>
              <td>
                <div
                  className="fd-popover"
                >
                  <div
                    onFocus={[Function]}
                    onMouseDown={[Function]}
                    onTouchStart={[Function]}
                  >
                    <div
                      className="fd-popover__control"
                    >
                      <button
                        className="fd-button--light sap-icon--vertical-grip"
                        onClick={[Function]}
                      />
                    </div>
                  </div>
                </div>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</div>
`;

exports[`GenericList Renders title and description 1`] = `
<div
  className="fd-panel fd-panel--no-background"
>
  <div
    className="fd-action-bar sc-kafWEX kXuvQ"
    data-e2e-id="toolbar"
  >
    <div
      className="fd-action-bar__header sc-iELTvK cWSWPl"
      data-e2e-id="toolbar-header"
    >
      <h1
        className="fd-action-bar__title"
      >
        title
      </h1>
      <p
        className="fd-action-bar__description"
      >
        description
      </p>
    </div>
    <div
      className="fd-action-bar__actions sc-cmTdod fVngQh"
      data-e2e-id="toolbar-actions"
    >
      <div
        className="fd-search-input"
      >
        <div
          className="fd-popover"
        >
          <div
            className="fd-popover__control"
          >
            <div
              aria-expanded={false}
              aria-haspopup="true"
              className="fd-combobox-control"
            >
              <div
                className="fd-input-group fd-input-group--after"
              >
                <input
                  className="fd-input"
                  onChange={[Function]}
                  onClick={[Function]}
                  onKeyPress={[Function]}
                  placeholder="Type your query"
                  style={Object {}}
                  type="text"
                  value=""
                />
                <span
                  className="fd-input-group__addon fd-input-group__addon--after fd-input-group__addon--button"
                >
                  <button
                    className="fd-button--light sap-icon--search"
                    onClick={[Function]}
                  />
                </span>
              </div>
            </div>
          </div>
          <div
            aria-hidden={true}
            className="fd-popover__body fd-popover__body--no-arrow"
          >
            <div
              className=""
            >
              <nav
                className="fd-menu"
              >
                <ul
                  className="fd-menu__list"
                >
                  <li
                    className="fd-menu__item"
                  >
                    No result
                  </li>
                </ul>
              </nav>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="fd-panel__body abc"
  >
    <div
      className="fd-panel sc-gGBfsJ eaQlFN"
    >
      <div
        className="fd-panel__body sc-hEsumM gUkXwJ"
      >
        <table
          className="fd-table sc-ktHwxA kmRrJU"
        >
          <thead>
            <tr />
          </thead>
          <tbody />
        </table>
        <p
          className="sc-cIShpX eszIwp"
        >
          There are no applications available
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`GenericList Renders with minimal props 1`] = `
<div
  className="fd-panel fd-panel--no-background"
>
  <div
    className="fd-action-bar sc-kafWEX kXuvQ"
    data-e2e-id="toolbar"
  >
    <div
      className="fd-action-bar__header sc-iELTvK cWSWPl"
      data-e2e-id="toolbar-header"
    >
      <h1
        className="fd-action-bar__title"
      >
        
      </h1>
    </div>
    <div
      className="fd-action-bar__actions sc-cmTdod fVngQh"
      data-e2e-id="toolbar-actions"
    >
      <div
        className="fd-search-input"
      >
        <div
          className="fd-popover"
        >
          <div
            className="fd-popover__control"
          >
            <div
              aria-expanded={false}
              aria-haspopup="true"
              className="fd-combobox-control"
            >
              <div
                className="fd-input-group fd-input-group--after"
              >
                <input
                  className="fd-input"
                  onChange={[Function]}
                  onClick={[Function]}
                  onKeyPress={[Function]}
                  placeholder="Type your query"
                  style={Object {}}
                  type="text"
                  value=""
                />
                <span
                  className="fd-input-group__addon fd-input-group__addon--after fd-input-group__addon--button"
                >
                  <button
                    className="fd-button--light sap-icon--search"
                    onClick={[Function]}
                  />
                </span>
              </div>
            </div>
          </div>
          <div
            aria-hidden={true}
            className="fd-popover__body fd-popover__body--no-arrow"
          >
            <div
              className=""
            >
              <nav
                className="fd-menu"
              >
                <ul
                  className="fd-menu__list"
                >
                  <li
                    className="fd-menu__item"
                  >
                    No result
                  </li>
                </ul>
              </nav>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="fd-panel__body abc"
  >
    <div
      className="fd-panel sc-gGBfsJ eaQlFN"
    >
      <div
        className="fd-panel__body sc-hEsumM gUkXwJ"
      >
        <table
          className="fd-table sc-ktHwxA kmRrJU"
        >
          <thead>
            <tr />
          </thead>
          <tbody />
        </table>
        <p
          className="sc-cIShpX eszIwp"
        >
          There are no applications available
        </p>
      </div>
    </div>
  </div>
</div>
`;
